projects/admin/src/app/shared/components/dynamic-form/dynamic-form.component.ts

```diff
@@ -0,0 +1,93 @@
+import { CommonModule } from '@angular/common';
+import { Component, Input, OnInit, inject } from '@angular/core';
+import { FormControl, FormGroup, ReactiveFormsModule, ValidatorFn, Validators } from '@angular/forms';
+import { MatButtonModule } from '@angular/material/button';
+import { MatInputModule } from '@angular/material/input';
+import { MatSlideToggleModule } from '@angular/material/slide-toggle';
+import { APIService } from '../../../../core/services/api.service';
+import { JSONSchema, SchemaInfo } from '../../model/json-schema';
+import { getFirstType, schemaInfo } from '../../model/schame';
+
+interface PropertyInformation {
+  name: string;
+  property: JSONSchema;
+}
+
+@Component({
+  standalone: true,
+  selector: 'app-dynamic-form',
+  imports: [MatInputModule, ReactiveFormsModule, MatSlideToggleModule, MatButtonModule, CommonModule],
+  templateUrl: './dynamic-form.component.html',
+  styleUrls: ['./dynamic-form.component.scss'],
+})
+export class DynamicFormComponent implements OnInit {
+  apiService = inject(APIService);
+  @Input() entityName: string = '';
+  @Input() value: unknown;
+
+  dynamicForm: FormGroup = new FormGroup({});
+  schemaInfo!: SchemaInfo;
+  propertiesInfo: PropertyInformation[] = [];
+
+  getFirstType = getFirstType;
+  createFormGroup() {
+    const formGroup = new FormGroup({});
+    for (const [propertyName, property] of Object.entries(this.schemaInfo.schema.properties)) {
+      const type = Array.isArray(property.type) ? property.type[0] : property.type;
+      if (propertyName !== 'id' && type !== 'array') {
+        this.propertiesInfo.push({
+          name: propertyName,
+          property: property,
+        });
+        const control = new FormControl(property.default, this.collectValidators(propertyName, property));
+        formGroup.addControl(propertyName, control);
+      }
+    }
+
+    this.dynamicForm = formGroup;
+  }
+
+  ngOnInit(): void {
+    this.schemaInfo = schemaInfo(this.entityName, this.apiService);
+    this.createFormGroup();
+    if (this.value) {
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
```

Comment:
it is not good practice to disable the eslint
but this class is super generic so it is ok here